packages/nimble-components/src/select/specs/IxD.md

```diff
@@ -0,0 +1,254 @@
+
+The client-user should be able to specify whether they want to display selection with or without a check mark.
+
+![ ](spec-images/select-11.png)
+
+#### Filter
+The client-user should be able to specify whether they want filtering available in the select component. The filter field appears in the menu.
+
+![ ](spec-images/select-12.png)
+
+### Behavior
+#### Minimum Width
+> NOTE: NEEDS DEV REVIEW
+
+The field button minimum width is three times the height of the component (to make components targetable on touch screens). The menu width is the same as the field button until reaching the minimum width (without truncating) of the longest string in the menu items.
+
+![ ](spec-images/select-13.png)
+
+#### Menu Height
+The menu height should be enough to display five to seven menu items before overflowing with a vertical scrollbar (not pictured).
+
+#### Overflow Text
+> NOTE: NEEDS DEV & VISUAL DESIGN REVIEW
+
+All text in the select component should truncate, not wrap.
+
+![ ](spec-images/select-14.png)
+
+#### Progress & Loading
+When displaying menu items in the menu lags or is expected to take longer than normal, the menu should indicate that it's loading. The "Loading…" display is not interactive.
+
+![ ](spec-images/select-15.png)
+
+> NOTE: IS THIS NEEDED? IF YES, NEEDS VISUAL DESIGN & TECH WRITER
+
+Menus with dynamically loading menu items may have a "Load more" button. Clicking the "Load more" button should resolve into the "Loading…" indication, then load more menu items and scroll the menu as needed.
+
+![ ](spec-images/select-16.png)
+
+#### Touch-Screen Devices
+On touch-screen devices, the select component should bring up the menu in a Nimble or native tray instead of a popover. 
+
+![ ](spec-images/select-17.png)
+
+### Mouse Interactions
+Field buttons are used to open the menu.
+
+![ ](spec-images/select-18.png)
+
+Hovering over a field button indicates interactivity.
+
+![ ](spec-images/select-19.png)
+
+Clicking anywhere on a field button opens (if closed) the menu. If the field button contains a valid value, selection is indicated on the associated menu item.
+
+![ ](spec-images/select-20.png)
+
+Hovering over a menu item in the menu indicates interactivity. 
+
+![ ](spec-images/select-21.png)
+
+Clicking anywhere on a menu item selects the item, displays the value in field button value, and closes the menu.
+
+![ ](spec-images/select-22.png)
+
+When the menu is open, clicking anywhere outside of a menu item (including the field button) closes the menu with no changes.
+
+![ ](spec-images/select-23.png)
+
+> NOTE: NEEDS VISUAL DESIGN FOR MULTIPLE FOCUS STATES?
+
+If the select component allows filtering, the filter bar has keyboard focus when the menu is opened.
+
+![ ](spec-images/select-24.png)
+
+> NOTE: NEEDS TECH WRITER FOR TEXT
+
+Typing into the filter bar filters the menu items in the menu (not pictured). The menu should indicate when there are no matches found. The "No matches" indicator is not interactive.
```

Comment:
Does it bear mentioning that the filter text can match anywhere in the text for each option?